HeadersImports.md

```diff
@@ -0,0 +1,68 @@
+# Headers and Imports
+
+## Introduction
+
+Follow this set of guidelines when creating header files and importing them. The
+guidelines are designed to support a wide range of build systems and usage scenarios.
+
+In this document, the term `library` refers to a buildable package. In CocoaPods, it's a CocoaPod.
+In Swift Package Manager, it's a library target.
+
+## Header File Types and Locations
+
+* *Public Headers* - Headers that define the library's API. They should be located in
+  `FirebaseFoo/Sources/Public`. Any additions require a minor version update. Any changes or
+  deletions require a major version update.
+
+* *Public Umbrella Header* - A single header that includes the full library's public API located at
+  `FirebaseFoo/Sources/Public/FirebaseFoo.h`. This header should be included in
+   [Firebase.h](CoreOnly/Sources/Firebase.h).
+
+* *Library Internal Headers* - Headers that are only used by the enclosing library. These headers
+  should be located among the source files. [Xcode](https://stackoverflow.com/a/8016333) refers to
+  these as "Project Headers".
+
+* *Private Headers* - Headers that are available to other libraries in the repo, but are not part
+  of the public API. These should be located in `FirebaseFoo/Sources/Private`.
+  [Xcode](https://stackoverflow.com/a/8016333) and CocoaPods refer to these as "Private Headers".
+  Note that we are deprecating the usage of CocoaPods `private_headers` and should instead
+  publish them with `preserve_paths` and access them with a repo-relative import.
+
+## Imports
+
+* *Headers within the Library* - Use a repo-relative path. The one exception is
+  that public header imports from other public headers should do an unqualified
+  import like `import "publicHeader.h"` to avoid colliding with the public module import.
+
+* *Headers within the Repo* - Import an umbrella header like
```

Comment:
"Headers within the Repo" isn't mentioned above. I think you mean "private headers from another library in the repo".

Also, nowhere else do you describe what a private umbrella header is, nor how to write one. A section about how to share internal APIs with other Firebase libraries would be helpful.

This paragraph should also mention that you need to add references to the private headers you're using to your CocoaPods `preserve_paths`. This would be a good place for your note about deprecating `private_headers`.

Reply:
Clarified